ext/spl/spl_directory.c

```diff
@@ -215,9 +215,16 @@ static inline void spl_filesystem_object_get_file_name(spl_filesystem_object *in
 			if (intern->file_name) {
 				efree(intern->file_name);
 			}
-			intern->file_name_len = spprintf(&intern->file_name, 0, "%s%c%s",
-			                                 spl_filesystem_object_get_path(intern, NULL),
-			                                 slash, intern->u.dir.entry.d_name);
+			char *path = spl_filesystem_object_get_path(intern, NULL);
```

Comment:
Decls should be at the top level of a block. The `spl_filesystem_object_get_path()` call should probably utilize the second parameter to not call `strlen()` below (as it should already be calculated).

Reply:
Fixed but "pull" bot closed the MR not sure why 

anyway I reopened it here with the changes
https://github.com/php/php-src/pull/3771